Classes/Domain/Validator/LanguageSkillValidator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+namespace PAGEmachine\Ats\Domain\Validator;
+
+use TYPO3\CMS\Extbase\Validation\Validator\AbstractValidator;
+
+/*
+ * This file is part of the PAGEmachine ATS project.
+ */
+
+/**
+ * LanguageSkillValidator
+ */
+class LanguageSkillValidator extends AbstractValidator
+{
+    public function isValid($languageSkill)
+    {
+
+        $error = false;
+
+        //Either dropdown language or text language must be set
+        if ($languageSkill->getLanguage() === null && empty($languageSkill->getTextLanguage())) {
+            $this->result->forProperty('language')->addError(new \TYPO3\CMS\Extbase\Validation\Error('You must enter a language, either via dropdown oder text field.', 1522232293));
+            $error = true;
+        }
+
+        //Level must be set
+        if ($languageSkill->getLevel() === null) {
```

Comment:
Same here wrt `empty()`.

Reply:
Can't agree on this one. $level is an integer, checking for null simply means "check if it is filled in".
It could carry a 0, which is not used in ATS, but would also mean "filled in".

Reply:
OK, I didn't have a deeper look at the type of this property. With scalar and return type hints in mind we should check if we can enforce this to int at a later point in time. In this case `0` means no expertise at all. Which bears the question why one would add a skill he has no level at but that's something to think about. ;-)